packages/cli/src/options/beaconNodeOptions/execution.ts

```diff
@@ -80,4 +84,18 @@ export const options: CliCommandOptions<ExecutionEngineArgs> = {
     type: "string",
     group: "execution",
   },
+
+  "jwt-id": {
+    description:
+      "An optional identifier to be included in the id field in the claims of the jwt tokens for authentication with EL client's rpc server hosting engine apis",
+    type: "string",
+    group: "execution",
+  },
+
+  "jwt-version": {
+    description:
+      "An option version string to be included in the clv field in the claims of the jwt tokens for authentication with EL client's rpc server hosting engine apis",
```

Comment:
I don't get the use case of this

- Why would a user set a arbitrary value here, and not set the actual version like how it is done for libp2p agent version and user-agent header?
https://github.com/ChainSafe/lodestar/blob/c50db8f7ae47198c76070487a6cfd0009d9bfd39/packages/cli/src/cmds/beacon/handler.ts#L199-L201
- Even if we assume this is a is a boolean value to set it or not, what's the point. As I see it, we either send this all the time or not at all. This is used to tell the EL about what CL is calling it, maybe to track it via metrics or add some kind of temporary workaround to fix a client bug or misbehavior (this is the use case for buider user agent as far as I know). Now if it is off by default, both those use case don't work.

Maybe I am missing the purpose of CLV but could also just consider to add it all the time and only disable it if `--private` flag is set
https://github.com/ChainSafe/lodestar/blob/c50db8f7ae47198c76070487a6cfd0009d9bfd39/packages/cli/src/cmds/beacon/options.ts#L119-L120

Reply:
> I don't get the use case of this
> 
> * Why would a user set a arbitrary value here, and not set the actual version like how it is done for libp2p agent version and user-agent header?
>   https://github.com/ChainSafe/lodestar/blob/c50db8f7ae47198c76070487a6cfd0009d9bfd39/packages/cli/src/cmds/beacon/handler.ts#L199-L201
> * Even if we assume this is a boolean value to set it or not, what's the point. As I see it, we either send this all the time or not at all. This is used to tell the EL about what CL is calling it, maybe to track it via metrics or add some kind of temporary workaround to fix a client bug or misbehavior (this is the use case for builder user agent as far as I know). Now, if it is off by default, both those use case don't work.
> 
> Maybe I am missing the purpose of CLV but could also just consider to add it all the time and only disable it if `--private` flag is set
> 
> https://github.com/ChainSafe/lodestar/blob/c50db8f7ae47198c76070487a6cfd0009d9bfd39/packages/cli/src/cmds/beacon/options.ts#L119-L120

I had the same doubt as you as well. I think having a user configurable version provides flexibility. 

I looked at lighthouse for reference, and they are just having a flag for [jwt-version](https://github.com/sigp/lighthouse/blob/stable/beacon_node/src/config.rs#L353). No mechanism to auto-populate this value.

Reply:
> > I had the same doubt as you as well. I think having a user configurable version provides flexibility.
> 
> What the difference to the ID then?
> 
> I guess right now CLV is unused at least as far I am aware but based on what the spec says, setting an arbitrary value sounds wrong
> 
> > The consensus layer client MAY use this to communicate the consensus layer client type/version.

I've reached out to LH team on discord about this, and imo their answer isn't too convincing.

I think we can have our own interpretation on the spec instead of mimicking LH's. 

How about we have `--include-jwt-id` and `--include-jwt-version` flags? If user passes in a value after the flag, Lodestar will include `id`/`clv` with the mentioned value. If user only adds the flag without any value, we populate default value eg. "" for `id`, and `Lodestar/${version}` for `clv`?

Reply:
I think we can just include the clv all the time. I haven't benchmarked it, but likely adding these few extra bytes in the payload will have a negligible effect.

As far as the id is concerned, using `--jwt-id`would match the naming of the existing jwt option. (Although that does differ from the camel casing of the other options, sadly). We probably shouldn't include an id in the payload unless the user has configured one.

Reply:
> As far as the id is concerned, using --jwt-id would match the naming of the existing jwt option.

We have changed flag names in the past and kept previous value as an alias, we could even have `jwt-id` as an alias for `jwtId` if we want to.

The only concern I can see if we rename `jwt-secret` to `jwtSecret` is if someone wants to look up what it does in the docs. I can't find a strong argument against renaming while having consistent naming patterns is not only cleaner UX but also makes them more intuitive, i.e. you can guess what the flag name might be without looking at docs, right now this is not the case for `jwt-secret`

> I think we can just include the clv all the time. I haven't benchmarked it, but likely adding these few extra bytes in the payload will have a negligible effect.

+1, I'd say either that or not include at all

Reply:
> > As far as the id is concerned, using --jwt-id would match the naming of the existing jwt option.
> 
> We have changed flag names in the past and kept previous value as an alias, we could even have `jwt-id` as an alias for `jwtId` if we want to.
> 
> The only concern I can see if we rename `jwt-secret` to `jwtSecret` is if someone wants to look up what it does in the docs. I can't find a strong argument against renaming while having consistent naming patterns is not only cleaner UX but also makes them more intuitive, i.e. you can guess what the flag name might be without looking at docs, right now this is not the case for `jwt-secret`
> 
> > I think we can just include the clv all the time. I haven't benchmarked it, but likely adding these few extra bytes in the payload will have a negligible effect.
> 
> +1, I'd say either that or not include at all

Updated!